web/test/test_base.js

```diff
@@ -0,0 +1,109 @@
+QUnit.module("base");
+QUnit.config.reorder = false;
+
+var Zstd = function () {
+
+  // public interface
+  return {
+
+    calls: [],
+
+    decompress: function (name, data) {
+      Zstd.calls.push({ decompress: [name, data.length] });
+    }
+  };
+}();
+
+
+function wait_for(assert, block) {
+  var done = assert.async();
+  return new Promise(resolve => {
+    // A function that checks the condition.
+    const check = () => {
+      // If the condition is met, resolve the promise.
+      var res;
+      try {
+        res = block();
+      } catch (ex) {
+        assert.ok(false, ex);
+      }
+      if (res) {
+        done();
+        resolve(res);
+      } else {
+        // schedule the next check.
+        requestAnimationFrame(check);
+      }
+    };
+
+    check();
+  });
+}
+
+async function load_image(num) {
+  const img = document.getElementById('example_frame' + num);
+  const imageBitmap = await window.createImageBitmap(img);
+  imageBitmap.requestVideoFrameCallback = function () { };
+
+  try {
+    Recv.init_video(imageBitmap);
```

Comment:
Very helpfully, the call we're using to extract information from a videoframe (which is where the camera feed ends up) also works perfectly well for an imagebitmap. So for our test, we can give it a few in succession to imitate pulling images from the camera.